src/NHibernate.Test/Linq/ODataTests.cs

```diff
@@ -53,6 +53,17 @@ public void OrderGroupBy(string queryString, int expectedRows)
 			Assert.That(results, Has.Count.EqualTo(expectedRows));
 		}
 
+		[TestCase("$filter=CustomerId le 'ANATR'",2 )]
+		[TestCase("$filter=startswith(CustomerId, 'ANATR')", 1)]
+		[TestCase("$filter=endswith(CustomerId, 'ANATR')", 1)]
+		[TestCase("$filter=indexof(CustomerId, 'ANATR') eq 0", 1)]
```

Comment:
Since we do not register `EndsWith`, `StartsWith` or `IndexOf` overloads with `StringComparison`, does this means there is an inconsistency in OData implementation here? It would emit `Compare` calls with that argument, but it would omit it for these three other cases?

So I see three options:
- only register what is currently required by OData, as does this PR currently.
- also register other cases overloads in case OData changes on this, or in case another Linq generator library requires them.
- reject this as an external issue, to be fixed on OData side.

@maca88, @hazzik

Reply:
> does this means there is an inconsistency in OData implementation here?

Yeap. Seems to be.

> also register other cases overloads in case OData changes on this, or in case another Linq generator library requires them.

I would choose this one as I see no difference with call to default `string.Compare`. It's not corelated in any way to with how DB is configured to make string comparisons. In fact `StringComparison` might be used to make LINQ to objects queries behave the same as DB (as I already mentioned in case of default SQL Server  case insensitive collation)

Reply:
Ok.

Reply:
By just adding the missing overload we introduce another issue where we generate the same query when using `StringComparison.XXX` and `StringComparison.XXXIgnoreCase`, which would cause an unexpected result on a case sensitive collation. IMO we should at least try to mimic .NET behavior when using `XXXIgnoreCase` by using `lower` function on each side.
> does this means there is an inconsistency in OData implementation here?

Yes, from the OData [source code](https://github.com/OData/WebApi/blob/813349a77669b5cd7c677538119b77f0ef82d755/src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs#L200) only `StringComparison.Ordinal` option is used for `string.Compare`, where for `EndsWith`, `StartsWith` and other methods the `StringComparison` overload [is not used](https://github.com/OData/WebApi/blob/d02bc61ea7b31ada1e54abbeebbecb3c5df0e3ac/src/Microsoft.AspNet.OData.Shared/Query/Expressions/ClrCanonicalFunctions.cs#L48).

Reply:
> By just adding the missing overload we introduce another issue where we generate the same query when using `StringComparison.XXX` and `StringComparison.XXXIgnoreCase`, which would cause an unexpected result on a case sensitive collation.

The thing is, we do not try to support different `StringComparison` values, we just ignore them. The trouble is the same conversely, on case insensitive collations, when using a non ignoring case `StringComparison`. And for this converse case, there is no ".Net" side workaround.

The current state is, NHibernate bluntly states these overloads are not supported. That is fine for user generated expression. But when the user cannot control the generated expression, it is a blocker.

So I think accepting these overloads, while not actually supporting the `StringComparison` argument, is an acceptable trade-of. Maybe we should emit a NHibernate warning or information about the argument being ignored, though.

Reply:
> And for this converse case, there is no ".Net" side workaround

Some databases supports the COLLATE clause ([SqlLite](https://www.sqlite.org/datatype3.html#collation), [SqlServer](https://blog.sqlauthority.com/2007/04/30/case-sensitive-sql-query-search/), [Oracle](https://oracle-base.com/articles/12c/column-level-collation-and-case-insensitive-database-12cr2#statement-level), [PostgreSql](https://www.postgresql.org/docs/current/collation.html), [MySql](https://dev.mysql.com/doc/refman/8.0/en/charset-collate.html)) within the query, which could be used to do case sensitive/insensitive comparison. But thats a matter for a separate PR.

> So I think accepting these overloads, while not actually supporting the StringComparison argument, is an acceptable trade-of

It is fine by me.